camPod/Classes/Photos/ViewModel/PhotosViewModel.swift

```diff
@@ -10,42 +10,87 @@ import Foundation
 public class PhotosViewModel {
     public weak var view: PhotoViewProtocol?
     public var repo: PhotosDatasourceProtocol?
-    
-    func loadPhotos(albumID: String, imagePaths: [String]) {
+    public var cacheHelper: CacheHelper?
+
+    func loadCachedImages(imagePaths: [String],
+                          _ completion: @escaping (_ imagePathsToUpdate: [String], _ cachedPhotoModels: [PhotoModel]) -> Void) {
+        var cachedPhotoModels = [PhotoModel]()
+        var imagePathsToUpdate = [String]()
         var count = 0
-        var tempImages = [UIImage]()
-        if imagePaths.contains("") {
+        for imagePath in imagePaths {
+            cacheHelper?.loadImageFromDisk(fileName: imagePath, { (imagePathToUpdate, cachedImage) in
+                count += 1
+                if let imagePathToUpdate = imagePathToUpdate {
+                    imagePathsToUpdate.append(imagePathToUpdate)
+                } else {
+                    cachedPhotoModels.append(PhotoModel(name: imagePath, image: cachedImage!))
+                }
+                if count == imagePaths.count {
```

Comment:
It seems like you're checking here if the loop is finished before the completion. Why not just perform the completion after the for loop ? Unless I'm missing something

Reply:
I found that the completion() wil run even it the loop does not finish which then cause the variable to be nil